Modules/CitiesViewController/CitiesViewController.swift

```diff
@@ -0,0 +1,248 @@
+        gradientView?.frame = tableView.bounds
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        tableView.bringSubviewToFront(rightBtn)
+    }
+    
+    // MARK: View Configurators
+    private func setupTableView() {
+        tableView.rowHeight = 60
+        tableView.showsVerticalScrollIndicator = false
+        tableView.backgroundColor = Colors.pageGradientFirstColor
+        tableView.register(
+            CityTableCell.self,
+            forCellReuseIdentifier: CityTableCell.identifier
+        )
+        //        tableView.bounces = false
+    }
+    private func addGradient() {
+        let pageGradient = Colors.pageGradient
+        
+        gradientView = tableView.setGradient(colors: pageGradient.colors,
+                                             locations: pageGradient.locations,
+                                             startPoint: pageGradient.startPoint,
+                                             endPoint: pageGradient.endPoint)
+    }
+    
+    private func setupPageTitleLbl() {
+        pageTitleLbl.setTitle("Cities")
+    }
+    
+    // MARK: Navigation Methods
+    private func presentWeatherDetailScreen(_ cityWeatherInfo: LocalStorageCity) {
+        let detailVC = CityWeatherEntriesVC(cityWeatherInfo: cityWeatherInfo)
+        navigationController?.pushViewController(detailVC, animated: true)
+    }
+    
+    private func setupRightBtn() {
+        /// Action
+        let addBtnAction: VoidCallback = { [unowned self] in
+            
+            let viewModel = SearchCityViewModel(weatherFetcher: WeatherFetcher(),
+                                                localStorageHelper: LocalStorageManager.shared)
+            
+            let didSelectCityCompletion: ((LocalStorageCity) -> Void) = { [unowned self] cityWeatherInfo in
+                
+                /// selected city ? dismiss search screen then present weather details screen
+                dismiss(animated: true)
+                
+                presentWeatherDetailScreen(cityWeatherInfo)
+                
+            }
+            
+            /// present search screen
+            let viewC = SearchCityViewController(viewModel: viewModel,
+                                                 openCityWeatherInfo: didSelectCityCompletion)
+            present(viewC, animated: true, completion: nil)
+        }
+        
+        let model = ReusableBtnModel(btnTappedAction: addBtnAction,
+                                     btnImage: UIImage(systemName: "plus"))
+        rightBtn.configureBtnWith(model)
+    }
+    
+    // MARK: SubViews
+    private lazy var rightBtn = ReusableButton()
+    private lazy var pageTitleLbl = ReusableBoldLabel()
+    private lazy var bottomImage = ReusableBottomImage(
+        frame: CGRect(
+            x: 0,
+            y: 0,
+            width: 0,
+            height: 0
+        )
+    )
+}
+
+
+// MARK: Constaints
```

Comment:
The programmatic constraints are hard to develop, read, and maintain. Please consider SnapKit as an alternative.